src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,234 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { getTimeOfNeedHealing } from '../../normalizers/CastLinkNormalizer';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import Events, { HealEvent, SummonEvent, DamageEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import LazyLoadStatisticBox from 'parser/ui/LazyLoadStatisticBox';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { SpellIcon, SpellLink } from 'interface';
+import { formatNumber } from 'common/format';
+import fetchWcl from 'common/fetchWclApi';
+import { WCLEventsResponse } from 'common/WCL_TYPES';
+import { Trans } from '@lingui/macro';
+import Combatants from 'parser/shared/modules/Combatants';
+
+//All possible results from a Time of Need proc
+enum Result {
+  Proc = 'Saved by Verdant Embrace',
```

Comment:
Not sure, but if it lets you do it you should use `<>Saved by<SpellLink spell={TALENTS_EVOKER.VERDANT_EMBRACE_TALENT} /></>`

Reply:
Doesn't seem to work, it just sends the whole thing as a string instead of making the SpellLink, maybe theres another way to handle the possible results?

Reply:
You could use an object instead of an enum to make it work eg.
```tsx
const Result = {
  Proc: (
    <>
      Saved by <SpellLink spell={TALENTS_EVOKER.VERDANT_EMBRACE_TALENT} />
    </>
  ),
  Long: 'Saved by all the healing',
  Not: 'Was not in danger of death',
  Dead: 'Died anyway',
  Bug: 'Time of Need fizzled out',
  Unparsed: "This event wasn't reviewed",
} as const;
//if you want a dynamic type; alternatively just "JSX.Element | string" it
type Results = typeof Result[keyof typeof Result];
```